pkg/controller/pod_volume_backup_controller.go

```diff
@@ -18,429 +18,333 @@ package controller
-	pvcLister             corev1listers.PersistentVolumeClaimLister
-	pvLister              corev1listers.PersistentVolumeLister
-	kbClient              client.Client
-	nodeName              string
-	metrics               *metrics.ServerMetrics
-	credentialsFileStore  credentials.FileStore
-
-	processBackupFunc func(*velerov1api.PodVolumeBackup) error
-	fileSystem        filesystem.Interface
-	clock             clock.Clock
-}
-
-// NewPodVolumeBackupController creates a new pod volume backup controller.
-func NewPodVolumeBackupController(
-	logger logrus.FieldLogger,
-	podVolumeBackupInformer informers.PodVolumeBackupInformer,
-	podVolumeBackupClient velerov1client.PodVolumeBackupsGetter,
-	podInformer cache.SharedIndexInformer,
-	pvcInformer corev1informers.PersistentVolumeClaimInformer,
-	pvInformer corev1informers.PersistentVolumeInformer,
-	metrics *metrics.ServerMetrics,
-	kbClient client.Client,
-	nodeName string,
-	credentialsFileStore credentials.FileStore,
-) Interface {
-	c := &podVolumeBackupController{
-		genericController:     newGenericController(PodVolumeBackup, logger),
-		podVolumeBackupClient: podVolumeBackupClient,
-		podVolumeBackupLister: podVolumeBackupInformer.Lister(),
-		podLister:             corev1listers.NewPodLister(podInformer.GetIndexer()),
-		pvcLister:             pvcInformer.Lister(),
-		pvLister:              pvInformer.Lister(),
-		kbClient:              kbClient,
-		nodeName:              nodeName,
-		metrics:               metrics,
-		credentialsFileStore:  credentialsFileStore,
-
-		fileSystem: filesystem.NewFileSystem(),
-		clock:      &clock.RealClock{},
-	}
-
-	c.syncHandler = c.processQueueItem
-	c.cacheSyncWaiters = append(
-		c.cacheSyncWaiters,
-		podVolumeBackupInformer.Informer().HasSynced,
-		podInformer.HasSynced,
-		pvcInformer.Informer().HasSynced,
-	)
-	c.processBackupFunc = c.processBackup
-
-	podVolumeBackupInformer.Informer().AddEventHandler(
-		cache.ResourceEventHandlerFuncs{
-			AddFunc:    c.pvbHandler,
-			UpdateFunc: func(_, obj interface{}) { c.pvbHandler(obj) },
-		},
-	)
-
-	return c
+// BackupExecuter runs backups.
+type BackupExecuter interface {
+	RunBackup(*restic.Command, logrus.FieldLogger, func(velerov1api.PodVolumeOperationProgress)) (string, string, error)
+	GetSnapshotID(*restic.Command) (string, error)
 }
 
-func (c *podVolumeBackupController) pvbHandler(obj interface{}) {
-	req := obj.(*velerov1api.PodVolumeBackup)
-
-	// only enqueue items for this node
-	if req.Spec.Node != c.nodeName {
-		return
-	}
-
-	log := loggerForPodVolumeBackup(c.logger, req)
-
-	if req.Status.Phase != "" && req.Status.Phase != velerov1api.PodVolumeBackupPhaseNew {
-		log.Debug("Backup is not new, not enqueuing")
-		return
-	}
-
-	c.metrics.RegisterPodVolumeBackupEnqueue(c.nodeName)
-
-	log.Debug("Enqueueing")
-	c.enqueue(obj)
+// PodVolumeBackupReconciler reconciles a PodVolumeBackup object
+type PodVolumeBackupReconciler struct {
+	Scheme         *runtime.Scheme
```

Comment:
What is the `Scheme` used for?

Reply:
I think Kubebuilder uses it (but I don't know what for) because it automatically adds Scheme as part of scaffolding a basic reconciler struct: https://book.kubebuilder.io/cronjob-tutorial/controller-overview.html